src/metal/scheduler.ts

```diff
@@ -144,29 +145,44 @@ export class PredicatedScheduler extends Scheduler implements SchedulerInterface
 
 export class ElementScheduler extends BaseScheduler implements ElementSchedulerInterface {
   protected queue: DOMQueue;
+  protected isDirty: boolean = false;
   constructor(customEngine?: EngineInterface) {
     super(customEngine);
     this.queue = new DOMQueue();
+    W.onWindowIsDirtyListeners.push({ fn: this.windowIsDirtyHandler, scope: this });
```

Comment:
rather then `scope` lets use `.bind`, as it is now supported and modern browsers have made it quite fast.

Reply:
This was intentional, type safety in TypeScript is lost when using `.bind` to pass scope. I'll dig a bit for alternatives. Also i'm almost certain that `.bind` is a bit slower than `.apply`.

Reply:
Ah, then an alternative approach would be in the constructor to do:

```
this.windowIsDirtyHandler = () => this.isDirty = true;
```

And then I believe we get:

* typesafety
* no extra pojo { } etc.

But I am open to @krisselden  recommendations.